### PR TITLE
Add localnet command

### PR DESCRIPTION
**Pull request by mationorato**
Add the ability to handle a local node from the `zksync-cli localnet` command, abstracting the user from previous knowledge of using Docker.

Requires https://github.com/matter-labs/local-setup/pull/8 to be merged first.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,6 +21,15 @@ You can install this program globally with `npm i -g zksync-cli` or run the comm
...
 > Both deposit and withdraw might take a couple of minutes to complete.
 
+- `zksync-cli localnet`: Manages a local zkSync 2.0 and Ethereum L1 testnet. It supports a set of sub-subcommands:
+  - `zksync-cli localnet up`: Bootstrap L1 and L2 localnets.
+  - `zksync-cli localnet down`: clear L1 and L2 localnets.
+  - `zksync-cli localnet start`: start L1 and L2 localnets.
+  - `zksync-cli localnet stop`: stop L1 and L2 localnets.
+  - `zksync-cli localnet logs`: Display logs.
+  - `zksync-cli localnet help`: Display this message and quit.
+  - `zksync-cli localnet wallets`: Display seeded wallet keys.
```
Comment (mm-zk) [inline README.md:31]: ❤

File: src/localnet.ts [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+// ---------------------------------------------------------------------------------------
+
+function runCommand(command: string, options?: ExecSyncOptions): string {
+    const defaultOptions: ExecSyncOptions = { cwd: repoDirectory(), encoding: 'utf-8' };
+    const unifiedOptions: ExecSyncOptions = {...defaultOptions, ...options};
+    return execSync(command, unifiedOptions).toString();
+}
+
+function repoDirectory(): string {
+    const xdgStateHome = process.env.XDG_STATE_HOME || path.join(os.homedir(), ".local/state");
```
Comment (mm-zk) [inline src/localnet.ts:20]: comment on why you use XDG_STATE_HOME

**Review by mm-zk — APPROVED** [review]

File: src/localnet.ts (hunk 2 above)
Comment (jpcenteno) [reply src/localnet.ts:20]: Sorry for the late response. I missed the notifications.

I decided to use `XDG_STATE_HOME` because I considered the presence or absence of this repo to be part of the CLI tool state.

I wanted to follow the  [XDG Base Directory Specification](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html) in order to follow some standard that keeps the end user's `$HOME` tidy.

For reference, here is the relevant part of the [XDG Base Directory Specification](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html)
>$XDG_STATE_HOME defines the base directory relative to which user-specific state files should be stored. If $XDG_STATE_HOME is either not set or empty, a default equal to $HOME/.local/state should be used.

The other alternative that I've considered was `XDG_DATA_HOME`
> XDG_DATA_HOME defines the base directory relative to which user-specific data files should be stored. If $XDG_DATA_HOME is either not set or empty, a default equal to $HOME/.local/share should be used.

I think that the other XDG base directories, `XDG_CONFIG_HOME`, `XDG_CACHE_HOME`, `XDG_RUNTIME_DIR` don't apply here.

What do you think?

Comment (mm-zk) [reply src/localnet.ts:20]: sounds good. please mention it in the code comment (so that the future readers can also understand this).

**Review by mm-zk — DISMISSED** [review]

**Comment by MexicanAce** [comment]
https://github.com/matter-labs/local-setup/pull/8 is now merged

File: bin/index.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
+#! /usr/bin/env node
```
Comment (MexicanAce) [inline bin/index.js:1]: Please remove the files in the `bin/` directory from this PR

File: package-lock.json [hunk 4]
```diff
@@ -1,12 +1,12 @@
 {
   "name": "zksync-cli",
-  "version": "0.1.4",
+  "version": "0.2.3",
```
Comment (MexicanAce) [inline package-lock.json:3]: We should revert these changes as we're moving to `semantic-release`

File: src/help.ts [hunk 5]
```diff
@@ -25,6 +25,10 @@ export default async function () {
   console.log(
     `Confirms the withdrawal of funds from zkSync to Layer 1. It will prompt for the network (localnet, testnet, mainnet), the transaction address of the withdrawal, and the private key of the wallet initiating the confirmation.\n`
   );
+  console.log(chalk.greenBright(`localnet`));
+  console.log(
+    `Manages a local zkSync 2.0 and Ethereum L1 testnet. Run zksync-cli localnet test for a list of supported operations.`
```
Comment (MexicanAce) [inline src/help.ts:30]: nit: Please refer to the network as `zkSync Era` for consistency

```suggestion
    `Manages a local zkSync Era and Ethereum L1 testnet. Run "zksync-cli localnet help" for a list of supported operations.`
```

File: src/localnet.ts (hunk 2 above)
Comment (MexicanAce) [reply src/localnet.ts:20]: As mentioned above, please add a comment with your explanation to the code

File: README.md [hunk 6]
```diff
@@ -25,6 +25,14 @@ You can install this program globally with `npm i -g zksync-cli` or run the comm
 
 > Both deposit and withdraw might take a couple of minutes to complete.
 
+- `zksync-cli localnet`: Manages a local zkSync 2.0 and Ethereum L1 testnet (it requires docker running on your system). It supports a set of sub-subcommands:
```
Comment (MexicanAce) [inline README.md:28]: For consistency, please refer to the network as `zkSync Era`:
```suggestion
- `zksync-cli localnet`: Manages a local zkSync Era and Ethereum L1 testnet (it requires docker running on your system). It supports a set of sub-subcommands:
```

File: src/localnet.ts (hunk 2 above)
Comment (jpcenteno) [reply src/localnet.ts:48]: Addressed @ [40b8aa8](https://github.com/matter-labs/zksync-cli/pull/16/commits/40b8aa856a316b421e689fe45752d15002704822)

File: bin/index.js (hunk 3 above)
Comment (jpcenteno) [reply bin/index.js]: Addressed @ [c58b62b](https://github.com/matter-labs/zksync-cli/pull/16/commits/c58b62b126d4298f5d68581deefb729c9dad8261)

**Comment by jpcenteno** [comment]
All comments and merge conflicts were addressed. I tested it manually on my workstation and it works as intended.

**Comment by jpcenteno** [comment]
I fixed the merge conflicts and adapted the PR to follow the new project structure introduced @ 632cb3614ca422469aee96d766e60692d538f32d (Nice job refactoring the project BTW)
